portfolio/src/main/webapp/script.js

```diff
@@ -16,121 +16,154 @@
+    modal.style.display = "block";
+    modalHeader.innerText = "Fun Facts";
+    modalImage.src = 'images/grantStarWars.jpg';
+    modalBody.innerText = addRandomFact();
+  }
+
+  mobileButton.onclick = function() {
+    modal.style.display = "block";
+    modalImage.src = 'images/grantShades.jpg';
+    modalHeader.innerText = "Mobile Development Projects";
+    modalBody.innerHTML = "DFG_Mobile".link("https://apps.apple.com/us/app/dfg-mobile/id1501790483?ign-mpt=uo%3D4");
+  }
+
+  webButton.onclick = function() {
+    modal.style.display = "block";
+    modalImage.src = 'images/grantFishing.jpg';
+    modalHeader.innerText = "Web Development Projects";
+    modalBody.innerHTML = "Morehouse Class of 2023 Website".link("https://morehouse23.info/");
+  }
+
+  githubButton.onclick = function() {
+    modal.style.display = "block";
+    modalHeader.innerText = "Github";
+    modalImage.src = 'images/grantStanding.jpg';
+    modalBody.innerHTML = "MyGithub Profile".link("https://github.com/GrantComm");
+  }
+
+  skillsButton.onclick = function() {
+    modal.style.display = "block";
+    modalHeader.innerText = "Skills";
+    modalImage.src = 'images/grantWalking.jpg';
+    modalBody.innerText = "Java, Webtechnologies, Python, C++, ....";
+  }
+
+  commentsButton.onclick = function() {
+    modal.style.display = "block";
+    modalImage.style.display = "none";
+    modalHeader.innerText = "Submit a Comment";
+    modalBody.innerText = "Comment";
+
+    var commentForm = document.createElement("form");
+    commentForm.setAttribute('method', "post");
+    commentForm.setAttribute('action', "/data");
+    var commentText = document.createElement("input");
+    commentText.setAttribute('type', "text");
+    commentText.setAttribute('name', "commentText");
+    var submitButton = document.createElement("input");
+    submitButton.setAttribute('type', "submit");
+    submitButton.setAttribute('value', "Submit");
+
+    commentForm.appendChild(commentText);
+    commentForm.appendChild(submitButton);
+
+    modalBody.appendChild(commentForm);
+  }
+
+  // When the user clicks anywhere outside of the modal, close it
+  window.onclick = function(event) {
+    if (event.target == modal) {
+      modal.style.display = "none";
     }
+  }
 }
 
 window.onscroll = function() { scrollFunction() };
 
 // Expands or contracts the navbar and displays to top button on scroll
 
 function scrollFunction() {
-    var mybutton = document.getElementById("myBtn");
-    var myNavBar = document.getElementById("nav");
-    // When the user scrolls down 20px from the top of the document, show the button
-    if (document.body.scrollTop > 310 || document.documentElement.scrollTop > 310) {
-        myNavBar.classList.add("extendedBar");
-        mybutton.style.display = "block";
-
-    } else {
-        mybutton.style.display = "none";
-        myNavBar.classList.remove("extendedBar");
-    }
+  var mybutton = document.getElementById("myBtn");
```

Comment:
Sorry I may have missed some earlier, but the style guide says not to use `var` at all because it's behavior is very weird (`var` variables are function-scoped). Please switch these to `let` instead.